builder/softlayer/client.go

```diff
@@ -182,9 +183,41 @@ func (self SoftlayerClient) doHttpRequest(path string, requestType string, reque
 		return nil, err
 	}
 
+	if err := self.hasErrors(decodedResponse); err != nil {
+		return nil, err
+	}
+
 	return decodedResponse, nil
 }
 
+func (self SoftlayerClient) doHttpRequestMany(path string, requestType string, requestBody *bytes.Buffer) ([]interface{}, error) {
+	responseBody, err := self.doRawHttpRequest(path, requestType, requestBody)
+	if err != nil {
+		err := errors.New(fmt.Sprintf("Failed to get proper HTTP response from SoftLayer API"))
+		return nil, err
+	}
+
+	var decodedResponse interface{}
+	err = json.Unmarshal(responseBody, &decodedResponse)
+	if err != nil {
+		err := errors.New(fmt.Sprintf("Failed to decode JSON response from SoftLayer: %s | %s", responseBody, err))
+		return nil, err
+	}
+
+	switch v := decodedResponse.(type) {
+	case []interface{}:
+		return v, nil
+	case map[string]interface{}:
+		if err := self.hasErrors(v); err != nil {
+			return nil, err
+		}
+
+		return nil, errors.New("Expected multiple results")
+	default:
+		return nil, errors.New("Expected multiple results")
+	}
+}
+
```

Comment:
If I understand this function correctly, the only difference between it and `doHttpRequest` is that in the `doHttpRequestMany` version you will throw an error if the response isn't an array.

I think it would be better to separate this validation logic into a separate function and use it in conjunction with `doHttpRequest`. 

What do you think ?